code/modules/surgery/organs/liver.dm

```diff
@@ -28,13 +28,16 @@
 			if(filterToxins)
 				//handle liver toxin filtration
 				var/toxamount
-				var/static/list/listOfToxinsInThisBitch = typesof(/datum/reagent/toxin)
-				for(var/datum/reagent/toxin/toxin in listOfToxinsInThisBitch)
-					toxamount += C.reagents.get_reagent_amount(initial(toxin.id))
-
+				var/static/list/listOfToxins = typesof(/datum/reagent/toxin)
+				for(var/I in C.reagents)
+					var/datum/reagent/pickedreagent = I
+					if(pickedreagent.type in listOfToxins)
+						toxamount += C.reagents.get_reagent_amount(initial(pickedreagent.id))
 				if(toxamount <= toxTolerance && toxamount > 0)
-					for(var/datum/reagent/toxin/toxin in listOfToxinsInThisBitch)
-						C.reagents.remove_reagent(initial(toxin.id), 1)
+					for(var/I in C.reagents)
+						var/datum/reagent/pickedreagent = I
+						if(pickedreagent.type in listOfToxins)
+						C.reagents.remove_reagent(initial(pickedreagent.id), 1)

```

Comment:
not a change request, but you don't *need* to use initial() here anymore as C.reagents contains instances. However, it is a small performance boost as "id" is nonvariable data and hence will be found on the prototype.

Reply:
So looking at the reagent to find it's initial then pulling the ID from that is cheaper than just looking at the reagent's ID?

Reply:
if you know that your data is stored on the type and not in the instance (ie. it's a number or a string, and has not been modified), then using `initial()` skips a step where byond checks the instance variables list looking for the variable definition